src/Controller/ComponentRegistry.php

```diff
@@ -85,9 +85,9 @@ public function setController(Controller $controller)
      * Part of the template method for Cake\Core\ObjectRegistry::load()
      *
      * @param string $class Partial classname to resolve.
-     * @return string|null Either the correct class name or null.
+     * @return string|object|null Either the correct class name or null.
      */
-    protected function _resolveClassName($class): ?string
+    protected function _resolveClassName($class)
```

Comment:
How can this return an object? The code does not show this.

Reply:
`ComponentRegistry` doesn't but there are other registry classes where this method can return object. Hence the return typehint is removed to match parent class.

Reply:
I didnt see any. If so, we might want to fix that instead maybe?

Reply:
`ConnectionRegistry` is one such registry. I don't think removing the ability to return objects would be an acceptable BC break.